virttest/env_process.py

```diff
@@ -526,7 +527,11 @@ def validate_memory_resource():
         if (params.get("setup_ksm") == "yes" and
                 params.get("ksm_run", "1") == "1"):
             magnification = 1.2
-        free_mem = "%s KB" % memory.freememtotal()
+        # Test whether is MemAvailable already introduced
+        if 'MemAvailable' in genio.read_file('/proc/meminfo'):
+            free_mem = "%s KB" % memory.read_from_meminfo('MemAvailable')
```

Comment:
Yes, it's not recommended to use for flow-control, but they are recommended to use to treat error statuses. The question is what do you see this as. It could be an error as the current system does not support "MemAvailable" and you have a fall-back "MemFree". So I'd be fine with exception in this case and the double read would only perform when the intended value is not available.

PS: The exception is `avocado.utils.process.CmdError`